### ISSUE TITLE
bug: Problems with jellyfin

### ISSUE THREAD

**Issue by Druidblack** [FoxxMD/multi-scrobbler#209]
### Please check existing knowledge before opening an issue

- [X] I have [checked the FAQ](https://foxxmd.github.io/multi-scrobbler/docs/FAQ/) and [documentation](https://foxxmd.github.io/multi-scrobbler/docs/configuration/)
- [X] I have [searched through existing issues](https://github.com/FoxxMD/multi-scrobbler/issues?q=sort%3Aupdated-desc) and [discussions.](https://github.com/FoxxMD/multi-scrobbler/discussions)

### Describe the Bug

1) When playing singles, when playback reaches the end, the track freezes for 1 second. And then it does not reset until a new music playback starts.
![Безымянный](https://github.com/user-attachments/assets/2c9e62eb-47d8-4a8c-95c8-6a4ea4c8fda6)
![34545345](https://github.com/user-attachments/assets/71598204-64f2-4645-bdd3-437b363ca424)
![87844](https://github.com/user-attachments/assets/079ba938-8bb1-4c1f-80b1-d4577e84b333)
2)When using the webhook, you could choose to send only data about the music playback. With the API version, now when you create screensavers or movies (TV series), the program captures all the data about the question.
![3453535111](https://github.com/user-attachments/assets/721a97f4-1806-4b02-bcdd-f5f74104be08)
![345352222](https://github.com/user-attachments/assets/4471aab7-bb46-4d51-af56-a78f52bf25a8)
2.1) The program also captures the playback of thematic videos (which play in the background of a movie or TV series)
![3453535111](https://github.com/user-attachments/assets/a13b6c5c-abf0-425c-b0b8-17c0d6e9c47f)
3) Why does Polling mode appear? Is this a mistake?
![64643222](https://github.com/user-attachments/assets/51b79427-7e96-4f60-ad24-89ce884c6df2)
Screenshots with logs (when single tracks hang)
![87844](https://github.com/user-attachments/assets/501eb1aa-5104-4b5d-b9ea-5778f92c6cb3)
![11123213](https://github.com/user-attachments/assets/f8b12ea1-de79-4c1c-9a81-4e43341ef977)

### Platform

Docker

### Versions

multi-scrobbler pr-205
jellyfin 10.9.11

**Comment by FoxxMD** [comment]
Please try docker image `foxxmd/mutli-scrobbler:pr-210`, it should fix incorrect media type (non-audio) from being detected.

> The program also captures the playback of thematic videos (which play in the background of a movie or TV series)

This is interesting I assumed jellyfin would only return video info for the playing movie. If you still see this behavior occurring with `pr-210` image please try enable `logPayload` in `jellyfin.json` like this:

```json5
{
  "name": "MyJellyfinSource",
  "data": { /* ... */ },
  "options": {
    "logPayload": true
  }
}

This will make MS output JSON data the first time it sees new media from the jellyfin API

![image](https://github.com/user-attachments/assets/be303cf9-ab00-4017-8d03-084582e6e1b9)

Please post a copy of this data (actual text, not screenshots) and logs when the background video/music is encountered by MS

> When playing singles, when playback reaches the end, the track freezes for 1 second. And then it does not reset until a new music playback starts.

If this still occurs when using `pr-210` please provide some more detail...im not sure what is you are describing. It may have been caused by the erroneous video sessions. **Text** copies of all the logs when this behavior is occurring would be helpful.

> Why does Polling mode appear? 

This is normal. The Jellyfin source is now making API calls to the Jellyfin server (**polling**) instead of waiting for the webhook to send data.

**Comment by Druidblack** [comment]
I can't install the 210 version.
**Failed to deploy a stack: multi-scrobbler Pulling multi-scrobbler Error Error response from daemon: manifest for foxxmd/multi-scrobbler:pr-210 not found: manifest unknown: manifest unknown**

foxxmd/**mutli**-scrobbler:pr-210 - foxxmd/**multi**-scrobbler:pr-210

**Comment by FoxxMD** [comment]
It's still building :sweat_smile:

**Comment by FoxxMD** [comment]
Should be available now

**Comment by Druidblack** [comment]
With version 210, only music is captured. Videos and thematic videos are not captured.
There is a problem with the theme music. In fact, your program is working correctly. It captures the music being played. But from the point of view of common sense, this is wrong.

The problem with singles remains. A single song in the playlist is not added to the playlists after reaching the end. Can you test this behavior on your instance? Or is this the only problem I have?

For jellyfin, service information has appeared in the list
![1](https://github.com/user-attachments/assets/72266388-60d0-43cb-adfb-9929e508490d)

For Last fm (Client), the artist's name is missing from the list. And official information also appeared.

![2](https://github.com/user-attachments/assets/90b8a79f-cedf-4220-986e-e3c43fd80114)

**Comment by FoxxMD** [comment]
> It captures the music being played. But from the point of view of common sense, this is wrong.

So please enable `logPayload` as I mentioned and post the **text** logs from the captured output when this happens. I may be able to filter this out if I can get examples of the data.

> The problem with singles remains. A single song in the playlist is not added to the playlists after reaching the end. Can you test this behavior on your instance? Or is this the only problem I have?

I cannot reproduce this. **You need to post full, text logs** of the behavior occurring for me to be able to properly debug this. Screenshots of parts of the log are not enough.

**Comment by Druidblack** [comment]
I don't use jellyfin.json for jellyfin. I use env variables. How do I use env variables in docker to enable logPayload?

**Comment by FoxxMD** [comment]
It is not possible to enable `logPayload` with only ENVs. Please use the [docker instructions for storage](https://foxxmd.github.io/multi-scrobbler/docs/installation/#storage) to mount a host directory to `/config` and then use the sample [`jellyfin.json`](https://foxxmd.github.io/multi-scrobbler/docs/configuration/?configType=file#jellyfin) to setup your source.

**Comment by Druidblack** [comment]
I created jellyfin.json, took the config from the example, replaced the address, name and Api. After restarting the container, jf did not appear in the source.

[jellyfin.json.txt](https://github.com/user-attachments/files/17368192/jellyfin.json.txt)

**Comment by FoxxMD** [comment]
Probably you want to delete all of the other [config properties](https://json-schema.app/view/%23%2Fdefinitions%2FJellyApiSourceConfig/%23%2Fdefinitions%2FJellyApiData?url=https%3A%2F%2Fraw.githubusercontent.com%2FFoxxMD%2Fmulti-scrobbler%2Fmaster%2Fsrc%2Fbackend%2Fcommon%2Fschema%2Fsource.json) and make sure you remove any existing JF ENVs.

```json
[
  {
    "name": "MyJellyfin",
    "enable": true,
    "clients": [],
    "data": {
      "url": "http://192.168.1.161:8096",
      "user": "admin",
      "apiKey": "8898e74df610432ebf0084b8cda0ee09",
    },
    "options": {
      "logFilterFailure": "warn",
      "logPayload": true
    }
  }
]
```

**Comment by Druidblack** [comment]
[jellyfin.json.txt](https://github.com/user-attachments/files/17368838/jellyfin.json.txt)
I have corrected the settings.  But the jf source never appears. The config folder is correct. I still have lastfm in it.json and it works correctly. I have no idea why jf is not starting. Help.

**Comment by FoxxMD** [comment]
Please provide full logs (not screenshots) of MS's startup. I cannot help you if you do not help me by providing logs and data for what is happening.

**Comment by Druidblack** [comment]
[scrobble-2024-10-14.1.log](https://github.com/user-attachments/files/17368956/scrobble-2024-10-14.1.log)

**Comment by FoxxMD** [comment]
Look at the validation error that is given at the very beginning:

```
Error: The config entry at index 0 from jellyfin.json was not valid
    at ScrobbleSources.buildSourcesFromConfig (CWD/src/backend/sources/ScrobbleSources.ts:517:43)
    at <anonymous> (CWD/src/backend/index.ts:121:9)
caused by: Error: Json config was not valid. Please use schema to check validity.

Schema Error:
At: /options/logFilterFailure
Data: true
enum: #/properties/logFilterFailure/enum => must be equal to one of the allowed values [false, debug, warn]

Schema Error:
At: /options/logFilterFailure
Data: true
enum: #/properties/logFilterFailure/enum => must be equal to one of the allowed values [false, debug, warn]
```

`logfilterFailure` should be "warn" or "debug". [Reference the example I gave you above.](https://github.com/FoxxMD/multi-scrobbler/issues/209#issuecomment-2412051851)

**Comment by Druidblack** [comment]
[scrobble-2024-10-14.1.log](https://github.com/user-attachments/files/17369013/scrobble-2024-10-14.1.log)
I made a mistake in the configuration file. I specified logFilterFailure twice. I've fixed it now.

[
  {
    "name": "MyJellyfin",
    "enable": true,
    "clients": [],
    "data": {
      "url": "http://192.168.1.161:8096",
      "user": "admin",
      "apiKey": "8898e74df610432ebf0084b8cda0ee09"
    },
    "options": {
      "logFilterFailure": "warn",
      "logPayload: "true"
    }
  }
]


jf does not appear.

**Comment by FoxxMD** [comment]
```
[2024-10-14 23:01:36.752 +0300] ERROR  : [App] [Sources] jellyfin.json config file could not be parsed
```

You have an error in your file. `logPayload` is missing a double quote.

```
"logPayload: "true"
"logPayload": "true"
```

If you are not using something like VSCode with built-in validation to check your files then consider using an online validator https://jsonlint.com/

**Comment by Druidblack** [comment]
In the message above, I attached a new debt file with the correction of the missing quotation mark. jf did not show up.

**Comment by FoxxMD** [comment]
Are you reading the logs you are sending?

```
Error: The config entry at index 0 from jellyfin.json was not valid
    at ScrobbleSources.buildSourcesFromConfig (CWD/src/backend/sources/ScrobbleSources.ts:517:43)
    at <anonymous> (CWD/src/backend/index.ts:121:9)
caused by: Error: Json config was not valid. Please use schema to check validity.

Schema Error:
At: /options/logPayload
Data: true
type: #/properties/logPayload/type => must be boolean

Schema Error:
At: /options/logPayload
Data: true
type: #/properties/logPayload/type => must be boolean

Schema Error:
At: 
Data: Object named 'MyJellyfin'
anyOf: #/anyOf => must match a schema in anyOf
    at validateJson (CWD/src/backend/utils/ValidationUtils.ts:59:15)
    at ScrobbleSources.buildSourcesFromConfig (CWD/src/backend/sources/ScrobbleSources.ts:507:45)
    at <anonymous> (CWD/src/backend/index.ts:121:9)
```

```
"logPayload": "true"
"logPayload": true
```

**Comment by Druidblack** [comment]
I'm sorry for the inattention. I won. It's working now. I am attaching a log file. 

After launching the program, I launched the theme music so that you could explore it.  The song is called theme, but this is not necessarily the name (different programs for downloading theme tunes may call files by different names).

Also, this melody was a single playback. This will show an error when the song reaches the end and stops at 01 seconds and is not added to the list of questions.

[scrobble-2024-10-14.1.log](https://github.com/user-attachments/files/17369177/scrobble-2024-10-14.1.log)

**Comment by FoxxMD** [comment]
Your logs do not include any verbose or debug-level logs. Please turn on debug logging by using ENV `LOG_LEVEL=debug`

**Comment by Druidblack** [comment]
[scrobble-2024-10-14.1.log](https://github.com/user-attachments/files/17369282/scrobble-2024-10-14.1.log)

**Comment by FoxxMD** [comment]
There was an issue with how `logPayload` worked that I've fixed. Please pull the latest image (should display version `pr-201-bb132101b5...` ) and post logs when the "theme" and single playlist issues are occurring again, thanks

**Comment by Druidblack** [comment]
[scrobble-2024-10-15.1.log](https://github.com/user-attachments/files/17377079/scrobble-2024-10-15.1.log)

**Comment by FoxxMD** [comment]
Thank you, this is the information I needed. This is interesting...

The media being picked up is `/mnt/film/5 Centimeters Per Second (2007) [tmdbid-38142]/theme.mp3` but it does not look like you were watching **5 Centimeters Per Second** when the logs were captured, is this correct? Did you browse to the movie in the JF interface and then this happened? Can you tell me more about what you were doing at this time?

**Comment by Druidblack** [comment]
The music plays when I'm on the movie page. Log. The file includes only the theme.

Yes. One of the ways to get audio and video themes for my collection.
I'll check it out now.

**Comment by FoxxMD** [comment]
These fixes are in [0.8.6](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.6) (`latest` docker image)
